Improve compare-with-tags.rb

ekohl opened the pull request:
It now:
* Is callable from any directory
* Handles all sections with the whitelist key
* Prints single koji commands

The current output:

 # Packages not expected in katello-nightly-rhel5
 * [x] python-rhsm
 * [x] subscription-manager
```shell
koji remove-pkg katello-nightly-rhel5 python-rhsm subscription-manager
```

 # Packages not expected in katello-nightly-rhel7
 * [x] katello-host-tools
 * [x] katello-installer
 * [x] tfm-rubygem-foreman_pipeline
 * [x] tfm-rubygem-hammer_cli_import
```shell
koji remove-pkg katello-nightly-rhel7 katello-host-tools katello-installer tfm-rubygem-foreman_pipeline tfm-rubygem-hammer_cli_import
```

 # Packages not expected in katello-thirdparty-rhel7
 * [x] tfm-rubygem-anemone
 * [x] tfm-rubygem-qpid_messaging
 * [x] tfm-rubygem-robotex
```shell
koji remove-pkg katello-thirdparty-rhel7 tfm-rubygem-anemone tfm-rubygem-qpid_messaging tfm-rubygem-robotex
```

ehelms commented:
On Thu, Mar 15, 2018 at 7:00 AM, Ewoud Kohl van Wijngaarden <
notifications@github.com> wrote:

> It now:
>
>    - Is callable from any directory
>    - Handles all sections with the whitelist key
>    - Prints single koji commands
>
> The current output:
> Packages not expected in katello-nightly-rhel5
>
>    - python-rhsm
>    - subscription-manager
>
> koji remove-pkg katello-nightly-rhel5 python-rhsm subscription-manager
>
> False output, this are expected in katello-nightly-rhel5


> Packages not expected in katello-nightly-rhel7
>
>    - katello-host-tools
>    - katello-installer
>    - tfm-rubygem-foreman_pipeline
>    - tfm-rubygem-hammer_cli_import
>
> koji remove-pkg katello-nightly-rhel7 katello-host-tools katello-installer tfm-rubygem-foreman_pipeline tfm-rubygem-hammer_cli_import
>
>
katello-host-tools is expected in katello-nightly-rhel7

> Packages not expected in katello-thirdparty-rhel7
>
>    - tfm-rubygem-anemone
>    - tfm-rubygem-qpid_messaging
>    - tfm-rubygem-robotex
>
> koji remove-pkg katello-thirdparty-rhel7 tfm-rubygem-anemone tfm-rubygem-qpid_messaging tfm-rubygem-robotex
>
> These are expected within this tag. All rubygems Katello packages but does
not maintain the source code of.


> ------------------------------
> You can view, comment on, or merge this pull request online at:
>
>   https://github.com/theforeman/foreman-packaging/pull/2283
> Commit Summary
>
>    - Improve compare-with-tags.rb
>
> File Changes
>
>    - *M* rel-eng/compare-with-tags.rb
>    <https://github.com/theforeman/foreman-packaging/pull/2283/files#diff-0>
>    (18)
>
> Patch Links:
>
>    - https://github.com/theforeman/foreman-packaging/pull/2283.patch
>    - https://github.com/theforeman/foreman-packaging/pull/2283.diff
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman-packaging/pull/2283>, or mute the
> thread
> <https://github.com/notifications/unsubscribe-auth/AAR588I-HenyJj8BsgowM6_sW38mXPmZks5tekm-gaJpZM4Sr8NR>
> .
>



-- 
Eric D. Helms
Red Hat Engineering

ekohl commented:
>> Packages not expected in katello-thirdparty-rhel7
>> 
>>     - tfm-rubygem-anemone
>>     - tfm-rubygem-qpid_messaging
>>     - tfm-rubygem-robotex
>> 
> > koji remove-pkg katello-thirdparty-rhel7 tfm-rubygem-anemone tfm-rubygem-qpid_messaging tfm-rubygem-robotex
> These are expected within this tag. All rubygems Katello packages but does
> not maintain the source code of.

In 1f5fb1990b46e07f2b2b56473b57d28696277802 I removed thirdparty because it was empty after I removed 3 unused gems from it. Should it be reintroduced?

ekohl commented:
Added a commit to add packages to the tito whitelist.

ehelms commented:
On Thu, Mar 15, 2018 at 1:01 PM, Ewoud Kohl van Wijngaarden <
notifications@github.com> wrote:

> Packages not expected in katello-thirdparty-rhel7
>
> - tfm-rubygem-anemone
> - tfm-rubygem-qpid_messaging
> - tfm-rubygem-robotex
>
> koji remove-pkg katello-thirdparty-rhel7 tfm-rubygem-anemone
> tfm-rubygem-qpid_messaging tfm-rubygem-robotex
> These are expected within this tag. All rubygems Katello packages but does
> not maintain the source code of.
>
> In 1f5fb19
> <https://github.com/theforeman/foreman-packaging/commit/1f5fb1990b46e07f2b2b56473b57d28696277802>
> I removed thirdparty because it was empty after I removed 3 unused gems
> from it. Should it be reintroduced?
>
Huh -- guess we stopped following this pattern I mentioned. You are correct
here. Carry on!


> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/theforeman/foreman-packaging/pull/2283#issuecomment-373449542>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAR58_MhGMdXrszZQ_DtKrFaywAlTchPks5tep5RgaJpZM4Sr8NR>
> .
>



-- 
Eric D. Helms
Red Hat Engineering
Ph.D. Student - North Carolina State University

ekohl commented:
Anything else that needs to happen?
